pandas/core/arrays/datetimelike.py

```diff
@@ -197,6 +201,27 @@ def astype(self, dtype, copy=True):
             return self._box_values(self.asi8)
         return super(DatetimeLikeArrayMixin, self).astype(dtype, copy)
 
+    # ------------------------------------------------------------------
+    # Formatting
+
+    def __repr__(self):
+        if len(self) <= 6:
```

Comment:
Since these are array-like, we could maybe piggyback on NumPy for these? https://docs.scipy.org/doc/numpy-1.14.0/reference/generated/numpy.set_printoptions.html

So get the threshold and edgeitems from `np.get_printoptions()['threshold']`? Though mixing two config systems sounds like a bad idea. Maybe we just add something to ours (can be done as a followup).

Reply:
> can be done as a followup

Yah, given the 1 week goal discussed in the call, a lot of this stuff I'm hoping to get something workable for now and follow-up when the pressure is off.  (same thought on the stacklevel check on the other PR)